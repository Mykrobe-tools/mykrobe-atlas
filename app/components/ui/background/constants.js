/* @flow */

export const SCALE = 1.5;

export const LOZENGES_PER_COLOR = 8;

export const LOZENGE_COLORS = [
  'rgba(0, 93, 138, 0.8)',
  'rgba(255, 137, 0, 0.7)',
];

export const LozengeDimensions = {
  width: SCALE * 185,
  height: SCALE * 65,
};
