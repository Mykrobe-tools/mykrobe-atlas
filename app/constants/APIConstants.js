/* @flow */

export const BASE_URL: string =
  process.env.NODE_ENV === 'production'
    ? (process.env.API_URL: any)
    : 'https://api.atlas-dev.makeandship.com';

// export const BASE_URL = 'https://localhost:3001';
// export const BASE_URL = 'http://13.69.243.89:8000';

export const USER_COOKIE_NAME = 'api.atlas.makeandship.com';
