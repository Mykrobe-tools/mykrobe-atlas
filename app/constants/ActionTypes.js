export const ANALYSE_FILE = 'ANALYSE_FILE';
export const ANALYSE_FILE_ERROR = 'ANALYSE_FILE_ERROR';
export const ANALYSE_FILE_SUCCESS = 'ANALYSE_FILE_SUCCESS';
export const ANALYSE_FILE_PROGRESS = 'ANALYSE_FILE_PROGRESS';
export const ANALYSE_FILE_CANCEL = 'ANALYSE_FILE_CANCEL';

export const SET_NODE_HIGHLIGHTED = 'SET_NODE_HIGHLIGHTED';
export const UNSET_NODE_HIGHLIGHTED_ALL = 'UNSET_NODE_HIGHLIGHTED_ALL';

export const LOAD_TREE = 'LOAD_TREE';
export const LOAD_TREE_SUCCESS = 'LOAD_TREE_SUCCESS';
