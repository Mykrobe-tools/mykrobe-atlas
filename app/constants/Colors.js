export const COLOR_TINT_PRIMARY = "#622268";
export const COLOR_TINT_SECONDARY = "#D06192";
export const COLOR_BASE_TEXT = "#676561";
export const COLOR_PLACEHOLDER_TEXT = "#ceccc6";
export const COLOR_BUFF = "#f7f6f1";
export const COLOR_BUFF_DARK = "#efede4";
export const COLOR_RULE = "#ceccc6";
export const COLOR_GREY_MID = "#9a9893";
