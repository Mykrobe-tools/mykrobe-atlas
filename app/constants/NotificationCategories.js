/* @flow */

export const ERROR = 'ERROR';
export const MESSAGE = 'MESSAGE';
export const SUCCESS = 'SUCCESS';
