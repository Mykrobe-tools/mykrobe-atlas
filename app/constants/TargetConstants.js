/* @flow */

export const TYPE_WEB = 'TYPE_WEB';
export const TYPE_DESKTOP = 'TYPE_DESKTOP';
export const SPECIES_S_AUREUS = 'SPECIES_S_AUREUS';
export const SPECIES_TB = 'SPECIES_TB';

export const TARGET_NAME = require('../../package.json').targetName;

let species;

if (TARGET_NAME === 'atlas-tb' || TARGET_NAME === 'desktop') {
  species = SPECIES_TB;
} else if (TARGET_NAME === 'atlas-s-aureus') {
  species = SPECIES_S_AUREUS;
} else {
  // target not recognised
  throw new Error(`Unsupported target name '${TARGET_NAME}'`);
}

export const SPECIES = species;
