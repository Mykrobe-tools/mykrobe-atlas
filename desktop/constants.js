export const DEBUG =
  process.env.DEBUG_PRODUCTION === '1' ||
  process.env.NODE_ENV === 'development';
