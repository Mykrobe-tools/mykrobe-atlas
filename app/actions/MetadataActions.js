/* @flow */

import fetchJson from '../api/fetchJson';
import { BASE_URL } from '../constants/APIConstants';
import * as ActionTypes from '../constants/ActionTypes';
import * as NotificationCategories from '../constants/NotificationCategories';
import {showNotification} from './NotificationActions';

export function postMetadataForm(id: string, metadata: Object) {
  return (dispatch: Function) => {
    return fetchJson(`${BASE_URL}/experiments/${id}`, {
      method: 'PUT',
      body: JSON.stringify(metadata)
    })
      .then((data) => {
        dispatch({
          type: ActionTypes.POST_METADATA_FORM
        });
        setTimeout(() => {
          dispatch(showNotification({
            category: NotificationCategories.SUCCESS,
            content: 'Metadata saved'
          }));
        }, 1000);
      })
      .catch((error) => {
        dispatch(showNotification({
          category: NotificationCategories.ERROR,
          content: error.statusText,
          autoHide: false
        }));
      });
  };
}

export function setMetadata(metadata: Object) {
  return {
    type: ActionTypes.SET_METADATA,
    metadata
  };
}

export function fetchTemplate(id: string) {
  return (dispatch: Function) => {
    const testData = require('../../test/_fixtures/api/template.json');
    dispatch({
      type: ActionTypes.SET_METADATA_TEMPLATE,
      template: testData
    });
  };
}
