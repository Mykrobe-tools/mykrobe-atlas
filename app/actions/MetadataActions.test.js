import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import nock from 'nock';
import { BASE_URL } from '../constants/APIConstants';
import * as MetadataActions from './MetadataActions';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('MetadataActions', () => {
  afterEach(() => {
    nock.cleanAll();
  });
  describe('postMetadataForm', () => {
    it('should create a "POST_METADATA_FORM" action', () => {
      const store = mockStore({});
      nock(BASE_URL)
        .put('/experiments/1')
        .reply(200, { status: 'ok' });
      jest.useFakeTimers();
      return store
        .dispatch(MetadataActions.postMetadataForm(1, {}))
        .then(() => {
          jest.runAllTimers();
          expect(store.getActions()).toMatchSnapshot();
        });
    });
  });

  describe('setMetadata', () => {
    it('should create a "SET_METADATA" action', () => {
      const metadata = {
        lorem: 'ipsum',
      };
      const expectedAction = {
        type: 'SET_METADATA',
        metadata,
      };
      expect(MetadataActions.setMetadata(metadata)).toEqual(expectedAction);
    });
  });
});
