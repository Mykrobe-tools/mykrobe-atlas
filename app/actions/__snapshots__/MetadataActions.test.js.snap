// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MetadataActions postMetadataForm should create a "POST_METADATA_FORM" action 1`] = `
Array [
  Object {
    "type": "POST_METADATA_FORM",
  },
  Object {
    "autoHide": false,
    "category": "ERROR",
    "content": undefined,
    "id": 0,
    "type": "SHOW_NOTIFICATION",
  },
]
`;
